### ISSUE TITLE
Rebase hashicorp/aws-sdk-go, migrate back to official AWS library

### ISSUE THREAD

**Issue by catsby** [hashicorp/terraform#1361]
Migrate Terraform back to upstream https://github.com/awslabs/aws-sdk-go

- [ ] ~~Rebase https://github.com/hashicorp/aws-sdk-go on `awslabs`~~ punted here
- [x] Test, make sure any patches to `hashicorp/aws-sdk-go` are applied or fixed in upstream
- [x] Migrate Terraform source back over to `awslabs/aws-sdk-go`

**Comment by jschneiderhan** [comment]
Once work is ready to start on this, I'd be happy to help root out the issues. My hope is that it decreases the frequency of the problems in #1051

**Comment by pmoust** [comment]
@catsby due to the way goland (does not) handle dependencies at the moment, wouldn't be safer to keep the hashicorp fork till this is resolved in language level?

**Comment by catsby** [comment]
>  due to the way goland (does not) handle dependencies at the moment, wouldn't be safer to keep the hashicorp fork till this is resolved in language level?

Can you clarify what you mean here?

**Comment by pmoust** [comment]
@catsby For instance currently `go get` fetches the current `HEAD` of git repos. If `awslabs` changes the codebase of `aws-sdk-go` our builds will fail or even worse if `terraform` does indeed build might not perform as expected. 
This applies to all dependencies that are not vendored (like with Godeps).

Some maintainers like @svanharmelen notifies us when there are breaking changes on cloudstack sdk he maintains for example, but this cannot on per person/project basis.

@mitchellh has mentioned in another issue that he is expecting golang to attach the dependency hell in the near future.

**Comment by pmoust** [comment]
How I manage to make so many grammatical/syntactical errors in 3 phrases is beyond me.. *..sigh..*

**Comment by catsby** [comment]
> If awslabs changes the codebase of aws-sdk-go our builds will fail or even worse if terraform does indeed build might not perform as expected. 

True, however that situation already exists for all of our dependencies at the moment; `awslabs` wouldn't be unique here. 

Dependency management is another pain here, yes. The Golang community wants to solve this better, and Mitchell is referring to an email chain on the golang email list. It's unlikely something will surface in the near future, but we can be hopeful :smile: In general I agree through. 

Step one is to rebase the changes they have made on to our fork of the library. There have likely been a good number of new features and fixes introduced. 

Step two can happen immediately after that too. 

Maybe we wait on Step 3, until we have a dependency management scheme in place. Like I mentioned though, we're already in that situation with our other dependencies.

**Comment by pmoust** [comment]
:+1:

**Comment by franklinwise** [comment]
How's this going?  Blocked by "Request Limit Exceeded"

**Comment by catsby** [comment]
Ah, this was completed last week and is in the master branch

**Comment by franklinwise** [comment]
Thank you @catsby works great now!!
